core/models.py

```diff
@@ -182,6 +182,9 @@ class Post(GenericModel):
     )
     display_counter = models.IntegerField(default=0)
     is_active = models.BooleanField(default=True)
+    is_draft = models.BooleanField(default=False)
+    is_archive = models.BooleanField(default=False)
+    is_published = models.BooleanField(default=False)

```

Comment:
```suggestion
    # FIXME: those fields (is_active, is_draft, is...) need to be put into one field to avoid logic hell later on.
    is_active = models.BooleanField(default=True)
    is_draft = models.BooleanField(default=False)
    is_archive = models.BooleanField(default=False)
    is_published = models.BooleanField(default=False)

```

Please commit suggestion, I just want a comment there because those multiple fields will lead to very complicated logic in future. We need to properly desing this. Right now it could have is_published = true and is_draft = true, which does not make sense.